### DISCUSSION TITLE
Is there a way to organize my command line args like -

### DISCUSSION THREAD

**Discussion by myndxero** [vladmandic/sdnext#1127] [Q&A]
In my screenshot.  This obviously didn't work, using parenthesis, but I am wondering if there's a way, so that each argument has it's own line. I know I don't need this for Vlad's, but I'd rather ask here than "over there." Haha. I am also setting up Anapnoe's and Auto's again, though Vlad will continue to be my main for as long as he supports it. The icon looks so good on my taskbar...

![image](https://github.com/vladmandic/automatic/assets/113237671/9af21b70-0af8-42ff-8a7f-984f4545753b)

![image](https://github.com/vladmandic/automatic/assets/113237671/081e513d-4726-4b29-95d4-10530d21d536)

**Comment by brknsoul** [comment]
A little messy, but
```
set COMMANDLINE_ARGS=--update-check
set COMMANDLINE_ARGS=%COMMANDLINE_ARGS% --update-all-extensions
set COMMANDLINE_ARGS=%COMMANDLINE_ARGS% --argument3
set COMMANDLINE_ARGS=%COMMANDLINE_ARGS% --argument4
set COMMANDLINE_ARGS=%COMMANDLINE_ARGS% --and so on
```
will result in
`COMMANDLINE_ARGS=--update-check --update-all-extensions --arguement3 --arguement4 --and so on`

This way, you can comment on the line by using `&:: <comment>`
```
set COMMANDLINE_ARGS=--update-check &:: this does something
set COMMANDLINE_ARGS=%COMMANDLINE_ARGS% --update-all-extensions &:: this does something else
...
```

You might want to check what arguments are available in vlad's, since a lot of them have been moved to the Settings tab.

**Reply by myndxero** [reply]
> You might want to check what arguments are available in vlad's, since a lot of them have been moved to the Settings tab.

![image](https://github.com/vladmandic/automatic/assets/113237671/74fe5e47-2839-4eb8-bd33-42db3832a095)

My custom webui-user.bat for Vlad's is very simple by comparison.

**Comment by vladmandic** [comment]
@brknsoul answer is correct, i'd add one more thing - you can replace >50% of your args with simple `--models-dir "Q:\Stable diffusion\_models"`

**Comment by myndxero** [comment]
Thank you both for the responses. I figured I could just commandline_arg every one individually somehow and that might end up being the actual answer, but wasn't sure how. Though it's been ages since I coded anything, there's still that need to want to organize any form of code I guess.

@brknsoul I may end up doing what you described plus super helpful &::, I didn't know that was a thing.

I have one more question should you swing back by here as a matter of curiosity - I used to use an update-all script to handle extensions as part of my update script for auto. I wanted to take what I learned there and use it to update my entire repository folder where I just had random respositories backed up. My update-all wouldn't work there, I could never figure out the syntax to go 2 directories in before git pulls. Not my code, I don't remember where I picked this up, but my repository structure would be, for instance:

X:/_vladmandic/automatic/extensions/

so the script wouldn't work from the root folder.

![image](https://github.com/vladmandic/automatic/assets/113237671/280cf34e-87a4-44b9-9f26-87f82e13efe9)

Thank you both again. Almost makes me want to get back into coding... there was always something so satisfying about having clean code, organized, efficient, that did the function it was designed to do. I dunno. Dunno if Vlad feels that way, or if that's a common feeling for ACTUAL coders.

Side note: in my first programming class for C++, kids were learning cout and cin still in the second or third week :P I had written a 20k line text adventure game where you could pick from 4 different classes with a complete level up curve from 1-50 and a primitive skill tree.

**Reply by vladmandic** [reply]
onleliner from cmd.exe: `for /d %D in ("*") do git -C %D pull`
if running inside batch script, you need to double each `%`
no need to `cd` into folder as git supports `-C` to specify working folder.
and if you ever do need to do a cd, best to do a `pushd`/`pop` as if you ever end up in junction point or something like that, who knows where `cd ..` would take you?

and yes, i do :)
